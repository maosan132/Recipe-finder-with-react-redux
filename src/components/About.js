import React from 'react';

const About = () => (
  <div>
    Browse over 7,000 recipes.
    Discover quick and easy dinners,
    delicious dessert recipes,
    the latest food trends and more.
    What will you cook today?
  </div>
);

export default About;
