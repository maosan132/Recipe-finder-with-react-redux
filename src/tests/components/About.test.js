import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import About from '../../components/About';

Enzyme.configure({ adapter: new Adapter() });

describe('<About />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<About />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains About Us header', () => {
    render(
      <Router>
        <About />
      </Router>,
    );
    expect(screen.getByText('About Us')).toBeInTheDocument();
  });

  test('it contains p element with texts', () => {
    const wrapper = shallow(<About />);
    const paragraph = (
      <p>
        Browse over 7,000 recipes. Discover quick and easy dinners, delicious
        dessert recipes, the latest food trends and more. What will you cook
        today?
      </p>
    );
    expect(wrapper.containsMatchingElement(paragraph)).toBe(true);
  });
});
