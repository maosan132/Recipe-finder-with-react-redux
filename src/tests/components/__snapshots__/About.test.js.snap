// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h2
        className="title"
      >
        About Us
      </h2>
      <img
        className="img-fluid"
        src="about.jpg"
      />
      <p
        className="paragraph"
      >
        Browse over 7,000 recipes. Discover quick and easy dinners, delicious dessert recipes, the latest food trends and more. What will you cook today?
      </p>
    </div>
  </div>
</div>
`;
